tasks/copy_to_glacier/handler.py

```diff
@@ -115,42 +83,37 @@ def task(event: Dict[str, Union[List[str], Dict]], context: object) -> Dict[str,
     glacier_bucket = config.get(CONFIG_BUCKETS_KEY).get('glacier').get('name')
-    collection_url_path = collection.get(COLLECTION_URL_PATH_KEY)
     granule_data = {}
-    for granule_url in granule_urls:
-        filename = os.path.basename(granule_url)
-        if filename not in granule_data.keys():
-            granule_data[filename] = {'granuleId': filename, 'files': []}
-        granule_data[filename]['files'].append(
-            {
-                'path': config[CONFIG_FILE_STAGING_DIRECTORY_KEY],
-                'url_path': config.get(CONFIG_URL_PATH_KEY, config[CONFIG_FILE_STAGING_DIRECTORY_KEY]),
-                'bucket': get_bucket_name_for_filename(filename, collection.get('files', [])),
-                'filename': granule_url,
-                'name': granule_url
-            }
-        )
-        if should_exclude_files_type(granule_url, exclude_file_types):
-            print(f"Excluding {granule_url} from glacier backup because of collection configured {EXCLUDE_FILE_TYPES_KEY}.")
-            continue
-        source = get_source_bucket_and_key(granule_url)  # todo: Handle 'None' return value.
-        copy_granule_between_buckets(source_bucket_name=source[1],
-                                     source_key=source[2],
-                                     destination_bucket=glacier_bucket,
-                                     destination_key=f"{collection_url_path}/{filename}")
-        print(f"Copied {granule_url} into glacier storage bucket {glacier_bucket}.")
-
-    final_output = list(granule_data.values())
-    return {'granules': final_output, 'input': granule_urls}
-
+    copied_file_urls = []
+
+    # Iterate through the input granules (>= 0 granules expected)
+    for granule in granules_list:
+        granuleId = granule['granuleId']
+        if granuleId not in granule_data.keys():
+            granule_data[granuleId] = {'granuleId': granuleId, 'files': []}
+
+        # Iterate through the files in a granule object
+        for file in granule['files']:
+            source_name = file['name']
+            source_filepath = file['filepath']
+            if should_exclude_files_type(source_name, exclude_file_types):
+                print(f"Excluding {source_name} from glacier backup because of collection configured {EXCLUDE_FILE_TYPES_KEY}.")
+                continue
+            copy_granule_between_buckets(source_bucket_name=file['bucket'],
+                                         source_key=source_filepath,
+                                         destination_bucket=glacier_bucket,
+                                         destination_key=source_filepath)
+            copied_file_urls.append(file['filename'])
+            print(f"Copied {source_filepath} into glacier storage bucket {glacier_bucket}.")
+
+    return {'granules': granules_list, 'copied_to_glacier': copied_file_urls}
 
 # handler that is provided to aws lambda
 def handler(event: Dict[str, Union[List[str], Dict]], context: object) -> Any:
```

Comment:
Docs will need to be updated here and in the generated docs in readme

Reply:
:+1: